src/psyclone/tests/psyir/transformations/intrinsics/mms_base_trans_test.py

```diff
@@ -445,318 +292,305 @@ def test_apply_multi(fortran_reader, fortran_writer, tmpdir):
         "  real :: array(n,m)\n"
         "  real :: value1, value2\n"
         "  real :: result\n"
-        "  result = value1 + sum(array) * value2\n"
+        "  result = value1 + maxval(array) * value2\n"
         "end subroutine\n")
-    expected_decl = "  real :: sum_var\n"
-    expected_bounds = (
-        "  do i_1 = 1, m, 1\n"
-        "    do i_0 = 1, n, 1\n")
-    expected_result = "  result = value1 + sum_var * value2\n"
+    expected = (
+        "  result = TINY(result)\n"
+        "  do idx = 1, m, 1\n"
+        "    do idx_1 = 1, n, 1\n"
+        "      result = MAX(result, array(idx_1,idx))\n"
+        "    enddo\n"
+        "  enddo\n")
     psyir = fortran_reader.psyir_from_source(code)
     # FileContainer/Routine/Assignment/BinaryOperation(ADD)/
     # BinaryOperation(MUL)/IntrinsicCall
     node = psyir.walk(IntrinsicCall)[0]
-    trans = NamedTestTrans()
+    trans = Maxval2CodeTrans()
     trans.apply(node)
     result = fortran_writer(psyir)
-    assert expected_decl in result
-    assert expected_bounds in result
-    assert expected_result in result
+    assert expected in result
     assert Compile(tmpdir).string_compiles(result)
 
 
-def test_apply_dimension_1d(fortran_reader, fortran_writer, tmpdir):
+def test_apply_dimension_1d(fortran_reader):
     '''Test that the apply method works as expected when a dimension
     argument is specified and the array is one dimensional. This
     should be the same as if dimension were not specified at all.
-    What we care about here are the sum_var array and the generated
-    loop bounds.
+    What we care about here are the initialisation of the result
+    variable, the generated intrinsic (MAX) and the loop bounds.
 
     '''
     code = (
         "subroutine test(array,value1,value2)\n"
         "  real :: array(:)\n"
         "  real :: value1, value2\n"
         "  real :: result\n"
-        "  result = value1 + sum(array,dim=1) * value2\n"
+        "  result = value1 + maxval(array,dim=1) * value2\n"
```

Comment:
Remove the value1 and value2 from this expression to make it the simplest example of the transformationerror.

Reply:
Simplified. Also removed the comment explaining the hierarchy as we use a walk.